### DISCUSSION TITLE
How do I order a module to be run last in a session-scope of multiple directories?

### DISCUSSION THREAD

**Discussion by mchesler613** [pytest-dev/pytest-order#69] [Q&A] [answered]
How do I configure pytest-order to run a module in a directory  as the **_last_** module in a session scope?  Thank you.
e.g.
```
directory_one/test_module1.py
directory_one/test_module2.py (run last)
directory_one/test_module3.py

directory_two/test_module1.py
directory_two/test_module2.py
directory_two/test_module3.py
```

**Comment by mrbean-bremen** [comment, marked as answer]
If you just want to order this one module, and not order any tests inside it, you can add an order marker to the last test in that module (either `pytest.mark.order(-1)` or `pytest.mark.order("last")`):

*directory_one/test_module2.py*
```py
import pytest

def test_1():
    pass

def test_2():
    pass

@pytest.mark.order(-1)
def test_3():
    pass
```

and use the module order group scope (see also the last example in the [documentation](https://pytest-order.readthedocs.io/en/latest/configuration.html#order-group-scope)):
```
$ python -m pytest --order-group-scope=module
```

If you always want to order module-wise, you may add this option to `pytest.ini`:

*pytest.ini*
```ini
[pytest]
; order tests first inside modules, order whole modules afterwards 
addopts = --order-group-scope=module
```

If you don't add any other order markers, your tests will be executed in the same order as without ordering, except that all tests in `directory_one/test_module2.py` will be executed last.

There is currently no possibility to prevent ordering inside the module, so you have to put the marker on the test you want to be executed last inside the module (if the execution order matters at all).

**Reply by mchesler613** [reply]
Thank you so much. This works like a charm !!

**Reply by mrbean-bremen** [reply]
Glad to hear!
